google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+                  SearchGoogleAdsRequest.newBuilder()
+                      .setQuery(query)
+                      .setCustomerId(Long.toString(customerId))
+                      .setPageSize(PAGE_SIZE)
+                      .build());
+
+          // Iterates over all rows in all pages to get all customer clients under the specified
+          // customer's hierarchy.
+          for (GoogleAdsRow googleAdsRow : response.iterateAll()) {
+            CustomerClient customerClient = googleAdsRow.getCustomerClient();
+            // The customer client making the request will be returned in the
+            // search result with a level of 0.
+            if (customerClient.getLevel().getValue() == 0) {
+              // Store the root customer client, which is the first encountered customer client
+              // that has level of 0.
+              if (rootCustomerClient == null) {
+                rootCustomerClient = customerClient;
+              }
+              // The steps below map parent and children accounts. Continue here so that managers
+              // accounts exclude themselves from the list of their children accounts.
+              continue;
+            }
+
+            // For all level-1 (direct child) accounts that are manager accounts, the above
+            // query will be run against them to create a map of managers to their
+            // child accounts for printing the hierarchy afterwards.
+            List<CustomerClient> customerClientChildren = new ArrayList<>();
+            // Updates the customerClientChildren with the list of children accounts if there
+            // exists an entry in the map for the given manager.
+            if (customerIdsToChildAccounts.containsKey(customerId)) {
+              customerClientChildren = customerIdsToChildAccounts.get(customerId);
+            }
+            // Adds the current child account to the list of the given manager's child accounts,
+            // which will be empty if the manager has not yet been added to the map.
+            customerClientChildren.add(customerClient);
+            // Inserts or updates the map for the given manager with the new list of child accounts.
+            customerIdsToChildAccounts.put(customerId, customerClientChildren);
+            // Checks is the child account is a manager itself so that it can later be processed
+            // and added to the map if it hasn't been already.
+            if (customerClient.getManager().getValue()) {
+              // A customer can be managed by multiple managers, so to prevent visiting the same
+              // customer multiple times, we need to check if it's already in the map.
+              boolean alreadyVisited =
+                  customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
+              if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
+                unprocessedCustomerIds.add(customerClient.getId().getValue());
```

Comment:
Can make this list immutable by adding all manager accounts from the hierarchy (as retrieved in the first API call with the query update I suggested above). That might help make the code easier to understand & maintain.